src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestUtils.java

```diff
@@ -0,0 +1,86 @@
+package net.snowflake.ingest.streaming.internal;
+
+import static net.snowflake.ingest.utils.Constants.RESPONSE_ERR_ENQUEUE_TABLE_CHUNK_QUEUE_FULL;
+import static net.snowflake.ingest.utils.Constants.RESPONSE_ERR_GENERAL_EXCEPTION_RETRY_REQUEST;
+import static net.snowflake.ingest.utils.Constants.RESPONSE_SUCCESS;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.Map;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.connection.RequestBuilder;
+import net.snowflake.ingest.connection.ServiceResponseHandler;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.http.client.HttpClient;
+
+public class StreamingIngestUtils {
```

Comment:
We have very similar retry logic in `HttpUtil.java` for all Ingest APIs, could you add your logic there and consolidate (see `getServiceUnavailableRetryStrategy`)?

Reply:
I opted to separate out this logic because it interacts with streaming ingest specific status codes.  I didn't want to add unnecessary overhead to every non-streaming ingest api call.  The method `getServiceUnavailableRetryStrategy` is only invoked if the API returns a certain set of exceptional responses.  For the cases I'm dealing with here we already have a 200 response so that method wouldn't be invoked.  In order to make it work with that framework we'd need to add an interceptor that would convert our specific use cases (200 http code, but non-success streaming ingest specific code) into an IOException that would then cause `getServiceUnavailableRetryStrategy` to be invoked.